subsys/shell/backends/Kconfig.backends

```diff
@@ -625,4 +625,25 @@ config SHELL_DUMMY_INIT_LOG_LEVEL
 
 endif # SHELL_BACKEND_DUMMY
 
+config SHELL_BACKEND_ADSP_MEMORY_WINDOW
+	bool "Shell backend implemented over mapped memory window."
+	depends on SOC_FAMILY_INTEL_ADSP
+	help
+	  Enable ADSP memory window backend which can be used to execute commands
+	  from remote host processor that has access to the same memory window.
+
+config SHELL_BACKEND_ADSP_MEMORY_WINDOW_PROMPT
+	string "Displayed prompt name"
+	default "~$ "
+	help
+	  Displayed prompt name for ADSP memory window backend. If prompt is set,
+	  the shell will send two newlines during initialization.
+
+config SHELL_BACKEND_ADSP_MEMORY_WINDOW_POLL_INTERVAL
+	int "Poll interval (in microseconds)"
+	default 100
+	help
+	  This option can be used to modify the poll interval the backend uses
+	  to check for new commands.
+
```

Comment:
One last small remark. This shall be visible only if the backend is activated. Please take a look for example how the other backends are handled in the Kconfig:

It should be like:

```
if SHELL_BACKEND_ADSP_MEMORY_WINDOW

config SHELL_BACKEND_ADSP_MEMORY_WINDOW_PROMPT
 	string "Displayed prompt name"
 	default "~$ "
 	help
 	  Displayed prompt name for ADSP memory window backend. If prompt is set,
 	  the shell will send two newlines during initialization.

 config SHELL_BACKEND_ADSP_MEMORY_WINDOW_POLL_INTERVAL
 	int "Poll interval (in microseconds)"
 	default 100
 	help
 	  This option can be used to modify the poll interval the backend uses
 	  to check for new commands.endif # SHELL_BACKEND_ADSP_MEMORY_WINDOW

endif # SHELL_BACKEND_ADSP_MEMORY_WINDOW
```

You can also add